### DISCUSSION TITLE
Debugger needs a configuration

### DISCUSSION THREAD

**Discussion by antogilbert** [mrcjkb/rustaceanvim#221] [Q&A] [answered]
Hi, from the documentation I understood that if there was a DAP installed, rustaceanvim would pick it up automatically. Unfortunately i don't seem to see this happen. When i try to issue the command `:DapContinue` I get a message saying there is no configuration found for rust. The same message applies if I follow the instructions to configure a DAP that is not present in any of the distributions. What am I doing wrong?

This is my `:checkhealth rustaceanvim`:

```
 rustaceanvim: require("rustaceanvim.health").check()

Checking for Lua dependencies ~
- OK [mfussenegger/nvim-dap](https://github.com/mfussenegger/nvim-dap) installed.

Checking external dependencies ~
- OK rust-analyzer: found rust-analyzer 0.3.1831-standalone (39ad79bec 2024-02-04)
- OK Cargo: found cargo 1.75.0 (1d8b05cdd 2023-11-20)
- OK rustc: found rustc 1.75.0 (82e1608df 2023-12-21)
- OK debug adapter: found codelldb 

Checking config ~
- OK No errors found in config.

Checking for conflicting plugins ~
- OK No conflicting plugins detected.

Checking for tree-sitter parser ~
- OK tree-sitter parser for Rust detected.
```

**Comment by mrcjkb** [comment]
Hey 👋 

This plugin uses rust-analyzer to pick up debug configurations.
You can try the command `:RustLsp debuggables`.

If it doesn't enter a debug session, that means rust-analyzer wasn't able to find any debuggable applications.

**Reply by teenjuna** [reply]
Hey @mrcjkb, I just installed the for the debugging and also encountered this issue. `:RustLsp debuggables` indeed works, but is there a way to make it start using `dap.continue()`? Having one way to start a debugger in different languages would be great.

Btw, thanks for your work!

**Reply by mrcjkb** [reply, marked as answer]
Yes. You can enable it by setting `vim.g.rustaceanvim.dap.autoload_configurations` to `true`.

It is enabled by default for neovim >= 0.10 (nightly).
In neovim 0.9, it may block the UI while compiling the debug build.

**Reply by teenjuna** [reply]
Thanks, it works!
